src/wallet/db.cpp

```diff
@@ -511,6 +512,15 @@ void CDBEnv::CloseDb(const std::string& strFile)
 
 void CDBEnv::ReloadDbEnv()
 {
+    // Make sure that no Db's are in use
+    std::unique_lock<std::recursive_mutex> lock(cs_db);
```

Comment:
In commit "Replace cs_db with a recursive mutex use with a conditional_variable_any"

Would be good to add an AssertLockNotHeld(cs_db); above this, since it would be a bug if the mutex were held recursively when this was called (wait could hang if the lock wasn't released).

Reply:
Done